experimental/skills/DialogToDialog/DialogRootBot/Dialogs/SkillDialog.cs

```diff
@@ -0,0 +1,153 @@
+                    messageActivity.Text = dialogArgs.Text;
+                    skillActivity = (Activity)messageActivity;
+                    break;
+
+                default:
+                    throw new ArgumentException($"Invalid activity type in {dialogArgs.ActivityType} in {nameof(SkillDialogArgs)}");
+            }
+
+            ApplyParentActivityProperties(dc, skillActivity, dialogArgs);
+            return await SendToSkill(dc, skillActivity, skillInfo, cancellationToken);
+        }
+
+        public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken = default)
+        {
+            await dc.Context.SendTraceActivityAsync($"{GetType().Name}.ContinueDialogAsync(). ActivityType: {dc.Context.Activity.Type}", cancellationToken: cancellationToken);
+
+            var skillId = await _activeSkillProperty.GetAsync(dc.Context, () => null, cancellationToken);
+
+            if (dc.Context.Activity.Type == ActivityTypes.Message && dc.Context.Activity.Text.Equals("abort", StringComparison.CurrentCultureIgnoreCase))
+            {
+                // Send a message to the skill to let it do some cleanup
+                var eocActivity = Activity.CreateEndOfConversationActivity();
+                eocActivity.ApplyConversationReference(dc.Context.Activity.GetConversationReference(), true);
+                await SendToSkill(dc, (Activity)eocActivity, _skillsConfig.Skills[skillId], cancellationToken);
+
+                // End this dialog and return (we don't care if the skill responds or not)
+                await dc.Context.SendTraceActivityAsync($"{GetType().Name}.ContinueDialogAsync(). Cancelled", cancellationToken: cancellationToken);
+                return await dc.EndDialogAsync(cancellationToken: cancellationToken);
+            }
+
+            if (dc.Context.Activity.Type == ActivityTypes.EndOfConversation)
+            {
+                await dc.Context.SendTraceActivityAsync($"{GetType().Name}.ContinueDialogAsync(). Got EndOfConversation", cancellationToken: cancellationToken);
+                return await dc.EndDialogAsync(dc.Context.Activity.Value, cancellationToken);
+            }
+
+            // Just forward to the remote skill
+            return await SendToSkill(dc, dc.Context.Activity, _skillsConfig.Skills[skillId], cancellationToken);
+        }
+
+        private async Task<DialogTurnResult> SendToSkill(DialogContext dc, Activity activity, BotFrameworkSkill skillInfo, CancellationToken cancellationToken)
+        {
+            // Always save state before forwarding
+            // (the dialog stack won't get updated with the skillDialog and things won't work if you don't)
+            await _conversationState.SaveChangesAsync(dc.Context, true, cancellationToken);
+            var response = await _skillClient.PostActivityAsync(_botId, skillInfo, _skillsConfig.SkillHostEndpoint, activity, cancellationToken);
+            if (response.Status != 200)
```

Comment:
Should any successful status code  succeed, such as 201?

Reply:
Added a check for if (response.Status >= 200 && response.Status <= 299). So far in my testing I've been always getting 200 but this should do I think.

---
In reply to: [356233007](https://github.com/microsoft/BotBuilder-Samples/pull/2032#discussion_r356233007) [](ancestors = 356233007)